### ISSUE TITLE
Implement missing textdomain argument sniff

### ISSUE THREAD

**Issue by ernilambar** [10up/plugin-check#1020]
The `WordPress.WP.I18n.MissingArgDomain` sniff currently flags when a translation function is missing the text domain argument. However, its current severity level means it might not receive appropriate attention from plugin developers.

### Background
- When the domain argument is missing in WordPress translation functions, the function will default to using core translations
- This behavior works technically but is discouraged in plugin or theme
- In 99.9% of cases, this occurs because plugin authors are unaware they need to specify their text domain, not because they intentionally want to use core translations

### Proposal
Increase the severity level of this sniff from its current level to **7 (as a blocker in PCP)**.

### Suggested improvement provided from Plugins Team meeting

Compare plugin strings with a map of WordPress core strings to reduce false positives when checking for missing text domains.

For example:

```php
echo esc_html__( 'Submit' ): // WARNING: This string exists in WP core.
echo esc_html__( 'Example plugin specific string' ): // ERROR: This string is not in core.
```

**Comment by rodrigoprimo** [comment]
Is there a reason why you can't set `default` as one of the allowed text domains? If you do so, `WordPress.WP.I18n.MissingArgDomain` will behave as you are describing. The only difference is that it won't check if the string exists in WP core, but I'm not sure if it is desirable to implement this in a sniff, as I believe it will be necessary to maintain the list of WP core strings.

Consider the following file:

```php
<?php

// phpcs:set WordPress.WP.I18n text_domain[] my-slug,default
__( 'string' );
__( 'string', 'default' );
__( 'string', 'my-slug' );
__( 'string', 'another-slug' );
```

Here is the sniff output:

```
$ vendor/bin/phpcs -s --standard=WordPress --sniffs=WordPress.WP.i18n test.php

FILE: test.php
----------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 1 WARNING AFFECTING 2 LINES
----------------------------------------------------------------------------------------------------------------------------------------------------------------------
 4 | WARNING | Missing $domain parameter in function call to __(). If this text string is supposed to use a WP Core translation, use the "default" text domain.
   |         | (WordPress.WP.I18n.MissingArgDomainDefault)
 7 | ERROR   | Mismatched text domain. Expected 'my-slug' or 'default' but got 'another-slug'. (WordPress.WP.I18n.TextDomainMismatch)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 42ms; Memory: 12MB

```

**Comment by swissspidy** [comment]
IMO `__( 'string' );` should error, but `__( 'string', 'default' );` should be a warning (like today).

This way, if you intentionally want to use a core string (like e.g. Performance Lab is doing), then you're still aware of it (through the warning) but can ignore it (because it's not blocking). That's why we implemented it that way originally.
